### PR TITLE
test(api): add TxOut JSON tests and document case* deprecation

### PR DESCRIPTION
**Pull request by newhoggy**
## Summary

This PR adds comprehensive test coverage for the TxOut JSON instances and documents the deprecation of era case* functions.

### 1. Deprecation Notice for Era Case Functions
Adds documentation to `Cardano.Api.Era.Internal.Case` noting that `case*` functions are deprecated and will be removed in a future release. New code should use direct pattern matching on era witnesses instead.

### 2. Comprehensive TxOut JSON Tests
Implements extensive test coverage for the `ToJSON` and `FromJSON` instances of `TxOut`, adapted for the `Exp.IsEra` constraint on `ToJSON`.

**New test modules:**
- `Test.Gen.Cardano.Api.TxOut`: Specialized generators for TxOut with specific datum types (no datum, datum hash, supplemental, inline) and invalid JSON scenarios
- `Test.Cardano.Api.TxOut.Helpers`: Test utilities including JSON field assertions, parse failure validators, and datum equality checks
- `Test.Cardano.Api.TxOut.Json`: Main test module organizing all test suites
- `Test.Cardano.Api.TxOut.JsonRoundtrip`: Roundtrip property tests for Conway era (CtxTx and CtxUTxO)
- `Test.Cardano.Api.TxOut.JsonEdgeCases`: Edge case tests for supplemental datum behavior, null field handling, ToJSON output validation
- `Test.Cardano.Api.TxOut.JsonErrorCases`: Error case tests for conflicting datums, mismatched hashes, partial fields, invalid data

**Note:** Tests are limited to Conway era because:
- `ToJSON` now uses `Exp.IsEra` constraint (Conway/Dijkstra only)
- Dijkstra era is not yet fully supported by `shelleyBasedEraConstraints`

## Test Coverage

- **Roundtrip tests**: CtxTx and CtxUTxO contexts
- **Datum-specific tests**: No datum, datum hash, supplemental datum, inline datum
- **Error handling**: Conflicting datums, mismatched hashes, partial fields, invalid data, missing required fields
- **Edge cases**: Supplemental datum field behavior, null field handling, ToJSON output validation

# Changelog

```yaml
- description: |
    Add comprehensive JSON roundtrip and error tests for TxOut instances.
    Document deprecation of case* functions in Era.Internal.Case.
  type:
    - test
    - documentation
  projects:
    - cardano-api
    - cardano-api-gen
```

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/gen/Test/Gen/Cardano/Api/TxOut.hs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      pure $
+        object
+          [ "address" .= addr
+          , "value" .= val
+          , "inlineDatum" .= scriptDataToJson ScriptDataJsonDetailedSchema datum
+          ]
+    ]
+
+-- | Generate a TxOut for a specific era (using appropriate datum types)
+genTxOutForEra
```
Comment (Jimbo4350) [inline cardano-api/gen/Test/Gen/Cardano/Api/TxOut.hs]: Where is this being consumed? This function is promoting the propagation of the eon based types we are trying to get rid of e.g `AlonzoEraOnwardsAlonzo` etc.
